File: docs/breez-sdk/snippets/csharp/ExternalSigner.cs
using Breez.Sdk.Spark;

namespace BreezSdkSnippets
{
    public class ExternalSignerSnippets
    {
        // ANCHOR: default-external-signer
        public static ExternalSigners CreateSigners()
        {
            var mnemonic = "<mnemonic words>";
            var network = Network.Mainnet;
            uint accountNumber = 0;

            var signers = BreezSdkSparkMethods.DefaultExternalSigners(
                mnemonic: mnemonic,
                passphrase: null,
                network: network,
                accountNumber: accountNumber
            );

            return signers;
        }
        // ANCHOR_END: default-external-signer

        // ANCHOR: connect-with-signer
        public static async Task<BreezSdk> ConnectWithSigner(ExternalSigners signers)
        {
            // Create the config
            var config = BreezSdkSparkMethods.DefaultConfig(Network.Mainnet) with
            {
                apiKey = "<breez api key>"
            };

            // Connect using the external signers
            var sdk = await BreezSdkSparkMethods.ConnectWithSigner(new ConnectWithSignerRequest(
                config: config,
                breezSigner: signers.breezSigner,
                sparkSigner: signers.sparkSigner,
                storageDir: "./.data"
            ));

            return sdk;
        }
        // ANCHOR_END: connect-with-signer

        // ANCHOR: sdk-builder-with-signer
        public static async Task<BreezSdk> BuildWithSigner(ExternalSigners signers)
        {
            var config = BreezSdkSparkMethods.DefaultConfig(Network.Mainnet) with
            {
                apiKey = "<breez api key>"
            };

            var builder = SdkBuilder.NewWithSigner(
                config: config,
                breezSigner: signers.breezSigner,
                sparkSigner: signers.sparkSigner
            );
            // await builder.WithStorageBackend(storage: <your storage backend>);
            // await builder.WithSharedContext(<your shared context>);
            var sdk = await builder.Build();

            return sdk;
        }
        // ANCHOR_END: sdk-builder-with-signer

        // ANCHOR: sdk-builder-with-signing-only-signer
        public static async Task<BreezSdk> BuildWithSigningOnlySigner(Config config, SigningOnlyExternalSigners signers)
        {
            var builder = SdkBuilder.NewWithSigningOnlySigner(
                config: config,
                breezSigner: signers.breezSigner,
                sparkSigner: signers.sparkSigner
            );
            var sdk = await builder.Build();

            return sdk;
        }
        // ANCHOR_END: sdk-builder-with-signing-only-signer
    }
}

File: docs/breez-sdk/snippets/python/src/external_signer.py
from breez_sdk_spark import (
    default_config,
    default_external_signers,
    connect_with_signer,
    BreezSdk,
    Config,
    ConnectWithSignerRequest,
    ExternalSigners,
    Network,
    SdkBuilder,
    SigningOnlyExternalSigners,
)

# ANCHOR: default-external-signer
def create_signers() -> ExternalSigners:
    mnemonic = "<mnemonic words>"
    network = Network.MAINNET
    account_number = 0

    signers = default_external_signers(
        mnemonic=mnemonic,
        passphrase=None,
        network=network,
        account_number=account_number,
    )

    return signers
# ANCHOR_END: default-external-signer

# ANCHOR: connect-with-signer
async def example_connect_with_signer(signers: ExternalSigners) -> BreezSdk:
    # Create the config
    config = default_config(Network.MAINNET)
    config.api_key = "<breez api key>"

    # Connect using the external signers
    sdk = await connect_with_signer(ConnectWithSignerRequest(
        config=config,
        breez_signer=signers.breez_signer,
        spark_signer=signers.spark_signer,
        storage_dir="./.data"
    ))

    return sdk
# ANCHOR_END: connect-with-signer

# ANCHOR: sdk-builder-with-signer
async def example_build_with_signer(signers: ExternalSigners) -> BreezSdk:
    config = default_config(Network.MAINNET)
    config.api_key = "<breez api key>"
    builder = SdkBuilder.new_with_signer(
        config=config,
        breez_signer=signers.breez_signer,
        spark_signer=signers.spark_signer,
    )
    # await builder.with_storage_backend(<your storage backend>)
    # await builder.with_shared_context(<your shared context>)
    sdk = await builder.build()
    return sdk
# ANCHOR_END: sdk-builder-with-signer

# ANCHOR: sdk-builder-with-signing-only-signer
async def example_build_with_signing_only_signer(
    config: Config, signers: SigningOnlyExternalSigners
) -> BreezSdk:
    builder = SdkBuilder.new_with_signing_only_signer(
        config=config,
        breez_signer=signers.breez_signer,
        spark_signer=signers.spark_signer,
    )
    sdk = await builder.build()
    return sdk
# ANCHOR_END: sdk-builder-with-signing-only-signer
